libantworld/agent.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+// OpenGL includes
+#include <GL/glew.h>
+
+// GLFW
+#include <GLFW/glfw3.h>
+
+// Libantworld includes
+#include "common.h"
+#include "renderer.h"
+
+// BoB robotics includes
+#include "../common/pose.h"
+#include "../video/opengl.h"
+
+using namespace BoBRobotics::Pose;
+
+namespace BoBRobotics {
+namespace AntWorld {
+template<class LengthUnit = meter_t, class AngleUnit = radian_t>
```

Comment:
I have yet to really look into the units stuff, but is it possible to template the set/get methods on the units passed to each function instead? That would mean that ``AntAgent`` itself can use use whatever units it wants internally and users of it can mix and match to match their mood? Having to have **every** class that uses units templated seems both annoying and a little restrictive...

Reply:
Yes we could do that and I think we probably should. Good shout.